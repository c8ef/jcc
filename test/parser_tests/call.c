int foo() { return 42; }

int bar(int x, int y) { return x + y; }

int main() {
  int x = foo();
  x = bar(12, 30);
  return x;
}
