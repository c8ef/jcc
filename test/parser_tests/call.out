FunctionDecl: foo
  Args[0]
  CompoundStatement
    ReturnStatement
      IntergerLiteral: 42
FunctionDecl: bar
  Args[2]
  VarDecl: x
  VarDecl: y
  CompoundStatement
    ReturnStatement
      BinaryExpr(+):
        DeclRefExpr: x
        DeclRefExpr: y
FunctionDecl: main
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
        CallExpr: foo
    ExprStatement
      BinaryExpr(=):
        DeclRefExpr: x
        CallExpr: bar
    ReturnStatement
      DeclRefExpr: x
